### PR TITLE
Fix PayPal Pay Later Parsing Error

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - Fix issue that causes a JSON parsing error when Pay Later is selected during checkout.

### Checklist

 - [x] Added a changelog entry
 - [x] Relevant test coverage

@sshropshire

### CODE REVIEW

File: PayPal/src/main/java/com/braintreepayments/api/PayPalCreditFinancingAmount.java [hunk 1]
```diff
@@ -22,12 +22,8 @@ private PayPalCreditFinancingAmount() {}
     static PayPalCreditFinancingAmount fromJson(JSONObject amount) {
         PayPalCreditFinancingAmount result = new PayPalCreditFinancingAmount();
 
-        if (amount == null) {
-            return result;
-        }
-
-        result.currency = Json.optString(amount, CURRENCY_KEY, null);
-        result.value = Json.optString(amount, VALUE_KEY, null);
```
Comment (sarahkoop) [inline PayPal/src/main/java/com/braintreepayments/api/PayPalCreditFinancingAmount.java:30]: Are these changes needed (and in native checkout too)? It looks like this logic would already tolerate the fields not existing

Comment (sshropshire) [reply PayPal/src/main/java/com/braintreepayments/api/PayPalCreditFinancingAmount.java:30]: The change we made falls back to empty strings when possible. I'm not sure if it's needed or not for MXO, but the change was easy enough to prevent a merchant filing an inbound and having to tell the merchant it's deprecated etc.

Comment (sarahkoop) [reply PayPal/src/main/java/com/braintreepayments/api/PayPalCreditFinancingAmount.java:30]: My concern with this is that now when the amount is null, the `PayPalCreditFinancingAmount` object returned will have the `currency` and `value` properties set to `""` rather than not set. That seems like unintended behavior

Comment (sshropshire) [reply PayPal/src/main/java/com/braintreepayments/api/PayPalCreditFinancingAmount.java:30]: Ok I see what you mean. I'll refactor the tests to expect `null` for those values.

Comment (sshropshire) [reply PayPal/src/main/java/com/braintreepayments/api/PayPalCreditFinancingAmount.java]: Made the update here: [8c95edc](https://github.com/braintree/braintree_android/pull/1059/commits/8c95edce819f3187c248d375bc0a9c41322e5013)

File: PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutCreditFinancingAmount.java [hunk 2]
```diff
@@ -25,10 +25,6 @@ private PayPalNativeCheckoutCreditFinancingAmount() {}
     static PayPalNativeCheckoutCreditFinancingAmount fromJson(JSONObject amount) {
         PayPalNativeCheckoutCreditFinancingAmount result = new PayPalNativeCheckoutCreditFinancingAmount();
 
-        if (amount == null) {
-            return result;
-        }
-
```
Comment (sarahkoop) [inline PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutCreditFinancingAmount.java:31]: Let's revert this too

Comment (sshropshire) [reply PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutCreditFinancingAmount.java:31]: Yeah we do own `Json.optString` though and it performs this check internally. I'm not sure why this guard exists, but I'm down to keep.

Comment (sshropshire) [reply PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutCreditFinancingAmount.java]: Done: [bb94c54](https://github.com/braintree/braintree_android/pull/1059/commits/bb94c54728c389230f80a07d57728e4265b1aebb)

**Review by sarahkoop — APPROVED** [review]

**Review by jaxdesmarais — APPROVED** [review]
